Additions

xDziugas opened the pull request:
Added correct, incorrect buttons functionality. Playing now works client-side. Added timesSolved and Ao3 (average of 3) properties to Quiz model to track recent average score and the amount of times quiz has been solved. resolved past conflicts

xDziugas committed:
playing works client-side, added correct/incorrect button functionallity and statistics (times played, average recent score)

xDziugas committed:
refactor

xDziugas committed:
resolved

xDziugas committed:
resolved

xDziugas committed:
Added Difficulty property to Question, lastPlayed property to Quiz. Small changes and fixes.

xDziugas commented:
- Added Difficulty property to Question (EASY, MEDIUM, HARD). 
- Added lastPlayed property to Quiz. 
- Quizzes in Library page are now being sorted by lastPlayed. 
- When playing, harder questions are always the last. 
- Refactored OnGet method in Playing/Index.cshtml.cs
- Minor changes to Playing page UI.

MrMnt commented on Models/Question.cs:
```diff
@@ -14,5 +14,13 @@ public class Question
         [ForeignKey("QuizId")]
         public int QuizId { get; set; }
         public string? UserId { get; set; } = null;
+        public Difficulty Difficulty { get; set; } = 0;
+    }
+
+    public enum Difficulty
+    {
+        EASY = 1,
+        MEDIUM,
+        HARD
     }
```
Create an enum value like NOT_GIVEN / NOT_ASSIGNED / NOT_AVAILABLE or something similar instead of setting to 0

MrMnt commented on Models/Quiz.cs:
```diff
@@ -21,5 +23,30 @@ public class Quiz
+
+        public DateTime? LastPlayed { get; set; } = null;
+
+        public int? Ao3Score { get; set; } = null;
```
Please change Ao3 abbreviation to something more clear

MrMnt replied:
Do we need to save Average Of 3 score to the database? If it's derived from past scores we can make it similar how you did with `public List<int> PastScores` and save the hassle of calculating it in other places like https://github.com/xDziugas/Acuedify/pull/16/files#diff-9fa5fb003bc6d204dbee4820ecadeed5a1341ecb3a096190f4ecaa187eb28ccdR139-R147

xDziugas committed:
small changes

xDziugas commented:
Changes:
- removed average score from the database
- added a method to LibraryService that calculates the average score for a given quizId
- adjusted the difficulty ENUM

xDziugas committed:
Merge branch 'main' into additions

xDziugas committed:
testing

xDziugas committed:
resolved conflicts
